testing/citest/tests/azure_smoke_test.py

```diff
@@ -0,0 +1,235 @@
+    self.TEST_SECURITY_GROUP = 'sec_grp_'+ self.TEST_APP
+    self.TEST_SECURITY_GROUP_RG = self.TEST_APP+ '-' + self.bindings['TEST_AZURE_RG_LOCATION']
+
+  #TODO: Perform the login to Azure with the SPN passed as parameters
+  def create_app(self):
+    """Creates OperationContract that creates a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_create_app_operation(
+            bindings=self.bindings, application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_AZURE_ACCOUNT']),
+        contract=contract)
+
+  def delete_app(self):
+    """Creates OperationContract that deletes a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_delete_app_operation(
+            application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_AZURE_ACCOUNT']),
+        contract=contract)
+
+  def create_a_security_group(self):
+    """Creates AzContract for createServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was created.
+    This will create a Network Security Group in a Resource Group on your azure Subscription
+    """
+    rules = [
+        {
+            "access": "Allow",
+            "destinationAddressPrefix": "*",
+            "destinationPortRange": "80-80",
+            "direction": "InBound",
+            "endPort": 80,
+            "name": self.TEST_SECURITY_GROUP,
+            "priority": 100,
+            "protocol": "tcp",
+            "sourceAddressPrefix": "*",
+            "sourcePortRange": "*",
+            "startPort": 80
+        }]
+    job = [{
+        "provider": "azure",
+        "application": self.TEST_APP,
+        "appName": self.TEST_APP,
+        "region": self.bindings['TEST_AZURE_RG_LOCATION'],
+        "stack": self.TEST_STACK,
+        "description": "Test - create security group for {app}".format(
+            app=self.TEST_APP),
+        "detail": "",
+        "credentials": self.bindings['SPINNAKER_AZURE_ACCOUNT'],
+        "securityRules": rules,
+        "name": self.TEST_SECURITY_GROUP,
+        "securityGroupName": self.TEST_SECURITY_GROUP,
+        "cloudProvider": "azure",
+        "type": "upsertSecurityGroup",
+        "user": "[anonymous]"
+    }]
+    builder = az.AzContractBuilder(self.az_observer)
+    (builder.new_clause_builder(
+        'Security Group Created', retryable_for_secs=30)
+     .collect_resources(
+         az_resource='network',
+         command='nsg',
+         args=['show', '--name', self.TEST_SECURITY_GROUP, '--resource-group', self.TEST_SECURITY_GROUP_RG],
+         no_resources_ok=True)
+     .contains_pred_list([
+         jp.DICT_MATCHES({
+             'name': jp.STR_SUBSTR(self.TEST_SECURITY_GROUP)#,
+             #'securityRules': jp.STR_SUBSTR("protocol='tcp'")
```

Comment:
I'm not sure what you were trying to do here.
Ideally your API returns back a json document so you have a dictionary with name value pairs (e.g. priority: 100) rather than a string (priority=100). If you have one big string you can either add a bunch of individual regexes into it or ideally have your agent be able to convert the string into a dictionary. I just checked in a python implementation of aws agent rather than CLI which is pretty simple. It's more simplistic than the gcp one. If your CLI cant give you json back then you might consider that route. It should be straightforward now that you've written the CLI one.

I think the multi regex solution here would look like
.contains_path_pred('securityRules',
    jp.AND(jp.STR_SUBSTR("protocol='tcp'"),
                 jp.STR_SUBSTR("priority=100"),
                 jp.STR_SUBSTR("destinationPortRange='80=80'")))

The reason a MATCH would be better, besides being simpler to read, is that the reporting is more straight forward, especially in the event of a failure.